### PR TITLE
Admin manage user + Search user

### PR DESCRIPTION
**Pull request by halo93**
User management includes listed tasks below:
- List all users
- Change role between user and admin
- Delete user
- Search user by email or name using ransack

Rubocop result:
Inspecting 44 files
............................................

44 files inspected, no offenses detected

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -222,9 +222,18 @@ GEM
...
+    kaminari-actionview (1.0.0)
+      actionview
+      kaminari-core (= 1.0.0)
+    kaminari-activerecord (1.0.0)
+      activerecord
+      kaminari-core (= 1.0.0)
+    kaminari-core (1.0.0)
     listen (3.0.8)
       rb-fsevent (~> 0.9, >= 0.9.4)
       rb-inotify (~> 0.9, >= 0.9.7)
```
Comment (nhunght) [inline Gemfile.lock]: checkout this file

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -22,10 +22,11 @@
...
-  $('#show-limit').on('change', function(){
-    $(this).closest('form').trigger('submit');
+  $(function(){
+    $('#show-limit').on('change', function(){
+      $(this).closest('form').trigger('submit');
+    });
   });
 });
+
+
```
Comment (nhunght) [inline app/assets/javascripts/application.js]: remove 2 dòng trắng này

File: app/controllers/admin/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    @q = User.search params[:q]
+    @users = @q.result(distinct: true)
+      .order_by_creation_time.page(params[:page]).per params[:limit]
+  end
+
+  def edit
+  end
+
+  def update
+    if @user.update_attributes role: user_params[:role]
```
Comment (nhunght) [inline app/controllers/admin/users_controller.rb]: cần ```@user.update_attributes user_params``` là đủ

File: app/views/admin/categories/index.html.erb [hunk 4]
```diff
@@ -26,4 +26,4 @@
       </tbody>
     </table>
     <%= paginate @categories %>
-<div>
+</div>
```
Comment (nhunght) [inline app/views/admin/categories/index.html.erb]: thiếu endfile

File: app/controllers/admin/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
+class Admin::UsersController < ApplicationController
+  load_and_authorize_resource param_method: :user_params, except: :update
```
Comment (nguyenthetoan) [inline app/controllers/admin/users_controller.rb]: không cần thiết phải khai báo param_method vì mặc định cancancan sẽ tìm theo tên model + params 
ref: https://github.com/CanCanCommunity/cancancan/wiki/Strong-Parameters#by-model-name

File: app/controllers/admin/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  before_action :authenticate_user!, :verify_admin
+  before_action :load_all_users, only: :index
+  before_action :load_user, only: [:edit, :update, :destroy]
```
Comment (nguyenthetoan) [inline app/controllers/admin/users_controller.rb]: cancancan sẽ load resource thông qua load_and_authorize_resource a nhé :)))
ref: https://github.com/CanCanCommunity/cancancan/wiki/authorizing-controller-actions

File: app/controllers/admin/users_controller.rb (hunk 5 above)
Comment (halo93) [inline app/controllers/admin/users_controller.rb]: Ok :+1:

File: app/controllers/admin/users_controller.rb (hunk 6 above)
Comment (halo93) [inline app/controllers/admin/users_controller.rb]: Ok :+1:

**Pull request merged**